nearcache_map.go

```diff
@@ -61,8 +77,19 @@ func newNearCacheMap(nc *nearCache, ncc *nearcache.Config, ss *serialization.Ser
 	return ncm, nil
 }
 
-func (ncm *nearCacheMap) registerInvalidationListener() {
-	fmt.Println("IMPLEMENT ME: registerInvalidationListener")
+func (ncm *nearCacheMap) registerInvalidationListener(ctx context.Context, name string, p *proxy, local bool) error {
+	// port of: com.hazelcast.client.map.impl.nearcache.NearCachedClientMapProxy#registerInvalidationListener
+	sid := types.NewUUID()
+	addMsg := codec.EncodeMapAddNearCacheInvalidationListenerRequest(name, eventTypeInvalidation, local)
+	handler := func(msg *proto.ClientMessage) {
+		switch msg.Type() {
+		case inearcache.EventIMapInvalidationMessageType:
+			ncm.handleInvalidationMsg(inearcache.DecodeInvalidationMsg(msg))
```

Comment:
Why don't we support batch invalidation? Also, 

where do we run this logic: (I think this one registers RepairingHandler in ReparingTask and returns it) https://github.com/hazelcast/hazelcast/blob/14ef0dbefa4a6fa00a50a8873aeea02ec6c3f5e4/hazelcast/src/main/java/com/hazelcast/client/map/impl/nearcache/NearCachedClientMapProxy.java#L673

Reply:
That's right, fixed at: https://github.com/hazelcast/hazelcast-go-client/pull/807/commits/adf231ea038a3c61a9ca428bc8a870b60458f291

Reply:
We support that, but not at the time of your comment :)

Reply:
It's at: nearcache_map.go:105 ( in `nearCacheMap.registerInvalidationListener` )